Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Lights/Ltc.azsli

```diff
@@ -80,14 +80,44 @@ float3x3 BuildViewAlignedOrthonormalBasis(in float3 normal, in float3 dirToView)
 // xy plane in positive z space.
 float IntegrateEdge(float3 v1, float3 v2)
 {
+    // This alternate version may work better for platforms where acos() precision is low.
+    /*
+    float x = dot(v1, v2);
+    float y = abs(x);
+
+    float a = 5.42031 + (3.12829 + 0.0902326 * y) * y;
+    float b = 3.45068 + (4.18814 + y) * y;
+    float theta_sinTheta = a / b;
+
+    if (x < 0.0)
+    {
+        theta_sinTheta = PI * rsqrt(saturate(1.0 - x * x)) - theta_sinTheta;
+    }
+
+    float3 u = cross(v1, v2);
+    return theta_sinTheta * u.z;
+    */
+
     float cosTheta = dot(v1, v2);
-    cosTheta = clamp(cosTheta, -0.9999, 0.9999);
+    float theta = acos(cosTheta);
 
     // calculate 1.0 / sin(theta)
-    float invSinTheta = rsqrt(1.0 - cosTheta * cosTheta);
+    float invSinTheta = rsqrt(saturate(1.0 - cosTheta * cosTheta));
 
-    float theta = acos(cosTheta);
-    return cross(v1, v2).z * theta * invSinTheta;
+    return cross(v1, v2).z * ((theta > 0.001) ? theta * invSinTheta : 1.0);
```

Comment:
that's a good add (specifically when it's below horizon).  I don't have the math link - why is a radian angle multiplied by a sin(angle) though?

Reply:
Math link: https://blog.selfshadow.com/publications/s2016-advances/s2016_ltc_rnd.pdf - As I understand it, this is from the original lambert integration of a polygon over a sphere.